### PR TITLE
[5] Feat/Dates: Dates as continuous value for ticks

### PR DESCRIPTION
**Pull request by Tanguylo**
Handle dates as continuous values on axes
image

image

### CODE REVIEW

**Comment by Tanguylo** [comment]
Still have a bug with precision => to wide zoom will give wrong ticks location, but label is well computed. This is hard to solve

File: plot_data/core.py [hunk 1]
```diff
@@ -45,6 +46,25 @@ def delete_none_from_dict(dict1):
...
+        if isinstance(element, list):
+            list_[i] = serialize_dates_in_list(element)
+        if isinstance(element, dict):
+            list_[i] = serialize_dates_in_dict(element)
+    return list_
+
+def serialize_dates_in_dict(dict_: Dict[str, Any]) -> Dict[str, Any]:
+    for (key, value) in dict_.items():
+        if isinstance(value, datetime.datetime):
+            dict_[key] =  f"{value.timestamp() * 1000}gmt+"
```
Comment (GhislainJ) [inline plot_data/core.py:62]: Consider setting this as a `serialize_date` function as it duplicated for lists and dicts

**Review by GhislainJ — COMMENTED** [review]
Ticks are now swapped vertically on all test images. Was it done on purpose  ?

File: cypress/e2e/axes.cy.ts [hunk 2]
```diff
@@ -124,7 +124,44 @@ describe('Axis', function() {
...
+      [
+        "07/03/1973 - 10:46:40",
+        "02/05/1976 - 21:33:20",
+        "05/07/1979 - 07:20:00",
+        "07/09/1982 - 17:06:40",
+        "03/11/1985 - 01:53:20",
+        "05/01/1989 - 11:40:00",
+        "07/03/1992 - 21:26:40"
+      ];
+    console.log(dateVector, dateAxis.labels, timeZoneOffSet)
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply cypress/e2e/axes.cy.ts:159]: Thanks

File: plot_data/core.py (hunk 1 above)
Comment (Tanguylo) [reply plot_data/core.py:62]: Changed

**Pull request merged**